artifactory.py

```diff
@@ -1569,7 +1569,13 @@ def __reduce__(self):
         # pathlib.PurePath.__reduce__ doesn't include instance state, but we
         # have state that needs to be included when pickling
         pathlib_reduce = super().__reduce__()
-        return pathlib_reduce[0], pathlib_reduce[1], self.__dict__
+        # pathlib.Path in 3.10+ does not have __slots__, so self.__dict__ is available;
+        # in 3.9 and earlier there is no self.__dict__. return slots instead
+        if sys.version_info.major == 3 and sys.version_info.minor >= 10:
```

Comment:
Hi! Could you add `_IS_PYTHON_3_10_OR_NEWER` constant for this case and use it? So we have all "python version related" constants in one place, like we'll do for 3.12 :pray: 
```python
_IS_PYTHON_3_10_OR_NEWER = sys.version_info >= (3, 10)
```

https://github.com/devopshq/artifactory/pull/463/files